ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java

```diff
@@ -0,0 +1,364 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud.azure;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.models.BlobContainerItem;
+import com.azure.storage.blob.models.BlobErrorCode;
+import com.azure.storage.blob.models.BlobHttpHeaders;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.blob.models.BlockBlobItem;
+import com.azure.storage.blob.options.BlobParallelUploadOptions;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.github.ambry.account.Container;
+import com.github.ambry.cloud.CloudBlobMetadata;
+import com.github.ambry.cloud.CloudBlobStoreV2;
+import com.github.ambry.cloud.CloudContainerCompactor;
+import com.github.ambry.cloud.CloudDestination;
+import com.github.ambry.cloud.CloudStorageException;
+import com.github.ambry.cloud.CloudUpdateValidator;
+import com.github.ambry.cloud.FindResult;
+import com.github.ambry.cloud.VcrMetrics;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.replication.FindToken;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class AzureCloudDestinationSync implements CloudDestination {
+
+  protected AzureBlobLayoutStrategy azureBlobLayoutStrategy;
+  protected AzureCloudConfig azureCloudConfig;
+  protected AzureMetrics azureMetrics;
+  protected BlobServiceClient azureStorageClient;
+  protected CloudConfig cloudConfig;
+  protected ClusterMap clusterMap;
+  protected ClusterMapConfig clusterMapConfig;
+  protected ConcurrentHashMap<String, BlobContainerClient> partitionToAzureStore;
+  protected VcrMetrics vcrMetrics;
+  public static final Logger logger = LoggerFactory.getLogger(AzureCloudDestinationSync.class);
+
+  /**
+   * Constructor
+   * @param verifiableProperties
+   * @param metricRegistry
+   * @param clusterMap
+   */
+  public AzureCloudDestinationSync(VerifiableProperties verifiableProperties, MetricRegistry metricRegistry, ClusterMap clusterMap) {
+    /**
```

Comment:
nit: May be `/*` instead of `/**` since latter is a java doc comment.

Reply:
done